pkg/parser/translate.go

```diff
@@ -419,21 +418,52 @@ func fromUDPIngressV1Alpha1(log logrus.FieldLogger, ingressList []*v1alpha1.UDPI
 			"udpingress_name":      ingress.Name,
 		})
 
-		result.ServiceNameToServices[ingress.Namespace+"."+ingress.Name] = kongstate.Service{
-			Service: kong.Service{
-				Name:     kong.String(ingress.Namespace + "." + ingress.Name),
-				Protocol: kong.String("udp"),
-				Host:     kong.String(ingressSpec.Host),
-				Port:     kong.Int(ingress.Spec.TargetPort),
-			},
-			Routes: []kongstate.Route{
-				{
-					Route: kong.Route{
-						Protocols:    []*string{kong.String("udp")},
-						Destinations: []*kong.CIDRPort{{Port: kong.Int(ingressSpec.ListenPort)}},
-					},
+		for i, rule := range ingressSpec.Rules {
+			// validate the ports and servicenames for the rule
+			if !util.IsValidPort(rule.Port) {
+				log.Errorf("invalid TCPIngress: invalid port: %d", rule.Port)
+				continue
+			}
+			if rule.Backend.ServiceName == "" {
+				log.Errorf("invalid UDPIngress: empty serviceName")
+				continue
+			}
+			if !util.IsValidPort(rule.Backend.ServicePort) {
+				log.Errorf("invalid UDPIngress: invalid servicePort: %d", rule.Backend.ServicePort)
+				continue
+			}
+
+			// generate the kong Route based on the listen port
+			route := kongstate.Route{
+				Ingress: util.FromK8sObject(ingress),
+				Route: kong.Route{
+					Name:         kong.String("udpingress" + "." + ingress.Namespace + "." + ingress.Name + "." + strconv.Itoa(i)),
```

Comment:
This is actually a break from what we do currently with TCPIngress, which just uses `ingress.Namespace + "." + ingress.Name + "." + strconv.Itoa(i)`.

We're in a weird place where the consistent option (use the same naming pattern for UDPIngress) poses a problem, because you could create conflicts by adding a TCPIngress and UDPIngress with the same name in the same namespace.

TCPIngress and Ingress didn't have this issue, but only because their rule counter scheme differed (Ingress counters have two digits because you can stuff multiple rules under a single hostname, whereas TCPIngress only has a single counter, as all rules have at most 1 hostname and port, same as UDPIngress). Ideally TCPIngress and UDPIngress would have similar naming schemes now, but that'd break backwards compatibility, albeit in a low-impact way.

Perhaps we should use `ingress.Namespace + "." + ingress.Name + "." + strconv.Itoa(i) + ".udp"`? IMO the suffix is a bit nicer since everything then uses `namespace.name` up front, and that's the more important part for a user trying to determine the source of route config.

Reply:
See https://github.com/Kong/kubernetes-ingress-controller/pull/1400#discussion_r649369884 for previous context about this change. Ultimately the suffix sounds good to me though :+1: 

```suggestion
					Name:         kong.String(ingress.Namespace + "." + ingress.Name + "." + strconv.Itoa(i)) + ".udp",
```